lib/ansible/plugins/connection/ssh.py

```diff
@@ -624,6 +747,33 @@ def exec_command(self, *args, **kwargs):
 
         return return_tuple
 
+    def _check_for_errors_in_return_results(self, return_code, stdout, stderr):
```

Comment:
Yup, very fragile. But given the myriad ways the ssh command can fail, we need to try to provide as much info to users as possible. If we can enhance the context provided with error messages in this scenario (where rc=255 and the ssh connection has failed without a doubt[1]) it should help cut down on user frustration. Especially given that current versions hide all of the ssh 

[1] Or more accurately, that the exec of the ssh client failed somehow, but it's about all we've got.